src/Integration/Binding/BindingController.cs

```diff
@@ -156,13 +156,16 @@ void IBindingWorkflowExecutor.BindProject(BindCommandArgs bindingArgs)
                 nugetBindingOp = new NoOpNuGetBindingOperation(host.Logger);
             }
 
+            var folderModifier = new Persistence.LegacySonarQubeFolderModifier(host);
```

Comment:
This looks weird. Why is the binding controller directly new-ing up the `LegacySonarQubeFolderModifier`?

If we always expect the same implementation of `ILegacySonarQubeFolderModifier` to be used and it just being being injected in the constructor for testing, then it would be better to have an _internal /* for testing */_ constructor, and have the public constructor call it a new instance.